pkg/csi/service/wcp/controller_helper.go

```diff
@@ -331,33 +331,43 @@ func getDatastoreURLFromStoragePool(ctx context.Context, spName string) (string,
 }
 
 // getAccessibleNodesFromStoragePool returns the accessibleNodes pertaining to the given StoragePool
-func getAccessibleNodesFromStoragePool(ctx context.Context, spName string) ([]string, error) {
+func getAccessibleNodesFromStoragePool(ctx context.Context, spName string) ([]string, string, error) {
 	// Get a config to talk to the apiserver
 	cfg, err := config.GetConfig()
 	if err != nil {
-		return nil, fmt.Errorf("failed to get Kubernetes config. Err: %+v", err)
+		return nil, "", fmt.Errorf("failed to get Kubernetes config. Err: %+v", err)
 	}
 
 	// create a new StoragePool client
 	spClient, err := dynamic.NewForConfig(cfg)
 	if err != nil {
-		return nil, fmt.Errorf("failed to create StoragePool client using config. Err: %+v", err)
+		return nil, "", fmt.Errorf("failed to create StoragePool client using config. Err: %+v", err)
 	}
 	spResource := spv1alpha1.SchemeGroupVersion.WithResource("storagepools")
 
 	// Get StoragePool with spName
 	sp, err := spClient.Resource(spResource).Get(ctx, spName, metav1.GetOptions{})
 	if err != nil {
-		return nil, fmt.Errorf("failed to get StoragePool with name %s: %+v", spName, err)
+		return nil, "", fmt.Errorf("failed to get StoragePool with name %s: %+v", spName, err)
 	}
 
 	// extract the accessibleNodes field
 	accessibleNodes, found, err := unstructured.NestedStringSlice(sp.Object, "status", "accessibleNodes")
 	if !found || err != nil {
-		return nil, fmt.Errorf("failed to find datastoreUrl in StoragePool %s", spName)
+		return nil, "", fmt.Errorf("failed to find datastoreUrl in StoragePool %s", spName)
 	}
 
-	return accessibleNodes, nil
+	// Get the storage pool type
+	storagePoolTypeMap, found, err := unstructured.NestedStringMap(sp.Object, "metadata", "labels")
+	if !found || err != nil {
+		return nil, "", fmt.Errorf("failed to find pool type in StoragePool %s", spName)
+	}
+	poolType, ok := storagePoolTypeMap["cns.vmware.com/StoragePoolType"]
```

Comment:
define cns.vmware.com/StoragePoolType as a constant on top. @shreesha / @subbu we need to start collecting all storage pool related constants in a common file that can be accessed from storage pool controller, placement engine and CSI.